### PR TITLE
Add WNP for Firefox 115 for VPN expansion [fix #13215]

### LINKED ISSUE #13215: WNP for Firefox 115: Mozilla VPN expansion
## Description

Firefox 115 releases on **4 July 2023**. Getting this page in early to allow time for l10n.

Promoting the launch of Mozilla VPN in a slew of new countries so this page will be localized in a number of languages. Some additional countries will still get the page, but in English, so we'll need to target geolocation in the template view separate from locale. Also some locales are pending translation of the VPN client, and may fall back to English if the client isn't localized at release time.

See the brief for more info, along with copy and design: https://docs.google.com/document/d/1MQfEleUcCDNAY_xAi3iqvSZgYccV5y7Q0T8jKjp4PSU/edit

There will be a traffic cop test to compare performance of two different coupon codes (to see if the text of the code makes much difference in conversion) with a 50/50 split between variants.

### PR DESCRIPTION
**Pull request by craigcook**
## One-line summary

Mozilla VPN is expanding to more countries. This adds a what's new page for Firefox 115 (releases 4 July, 2023) promoting Mozilla VPN in those countries: Bulgaria, Cyprus, Czech Republic, Denmark, Estonia, Croatia, Hungary, Lithuania, Luxembourg, Latvia, Malta, Poland, Portugal, Romania, Slovenia, and Slovakia.

It's controlled by the same switch that controls the entire expansion, `vpn-wave-vi`

Some will be localized but some will receive English content. We want to get strings out as soon as possible to allow time for l10n.

This also includes a traffic cop experiment testing two different coupon codes with a 50/50 split, controlled by the switch `experiment-firefox-whatsnew-115-eu-vpn`

## Issue / Bugzilla link

#13215 

## Testing

https://www-demo2.allizom.org/firefox/115.0/whatsnew/?geo=BG

http://localhost:8000/firefox/115.0/whatsnew/?geo=BG

The page is only shown in target countries so you'll need the `geo=` param to simulate it (or use a VPN). 
See `settings.VPN_COUNTRY_CODES_WAVE_VI` for the list of country codes (I used Bulgaria in the links above).

- [x] When the switch `vpn-wave-vi` is ON, target countries get the VPN promo page.
- [ ] When the switch `vpn-wave-vi` is OFF, target countries get the standard page.

### CODE REVIEW

File: l10n/en/firefox/whatsnew/whatsnew-115-vpn.ftl [hunk 1]
```diff
@@ -0,0 +1,18 @@
+# This Source Code Form is subject to the terms of the Mozilla Public
```
Comment (craigcook) [inline l10n/en/firefox/whatsnew/whatsnew-115-vpn.ftl:1]: @peiying2 Here are the strings for the VPN WNP.

**Comment by flodolo** [comment]
@craigcook 
Peiying is out, and I have no way to track down pings in GitHub.

Can you make sure to ping me instead in June? Are there are any other things I should look into?

File: l10n/configs/pontoon.toml [hunk 2]
```diff
@@ -213,6 +213,18 @@ locales = [
         "pt-PT",
         "nl",
     ]
+[[paths]]
+    reference = "en/firefox/whatsnew/whatsnew-115-vpn.ftl"
+    l10n = "{locale}/firefox/whatsnew/whatsnew-115-vpn.ftl"
+    locales = [
+        "cz",
```
Comment (flodolo) [inline l10n/configs/pontoon.toml:220]:
```suggestion
        "cs",
```

This ain't a locale code 😉

**Review by flodolo — APPROVED** [review]
The strings look good, the locale code for Czech in the TOML needs fixing

File: l10n/configs/pontoon.toml (hunk 2 above)
Comment (craigcook) [reply l10n/configs/pontoon.toml]: Oops! Fixed.

File: bedrock/firefox/views.py.orig
Comment (nathan-barrett) [inline bedrock/firefox/views.py.orig:1]: Whats this views.py.orig file do? I haven't seen these before

File: media/js/firefox/whatsnew/whatsnew-115-eu-vpn.js [hunk 3]
```diff
@@ -0,0 +1,25 @@
...
+        let successMsg = copyButton.dataset.success;
+
+        try {
+            await navigator.clipboard.writeText(couponCode);
+            copyButton.textContent = successMsg;
+        } catch (e) {
+            //do nothing
+        }
+    }
+})();
```
Comment (nathan-barrett) [inline media/js/firefox/whatsnew/whatsnew-115-eu-vpn.js:25]: look at you, using async/await. Javascript expert over here

File: media/js/firefox/whatsnew/whatsnew-115-experiment-eu-vpn.es6.js [hunk 4]
```diff
@@ -0,0 +1,41 @@
...
+                'data-ex-name': 'wnp-115-experiment-eu-vpn'
+            });
+        } else if (href.indexOf('v=2') !== -1) {
+            window.dataLayer.push({
+                'data-ex-variant': 'wnp115-coupon-code-vpn',
+                'data-ex-name': 'wnp-115-experiment-eu-vpn'
+            });
+        }
+    } else if (TrafficCop) {
+        const mcnulty = new TrafficCop({
```
Comment (nathan-barrett) [inline media/js/firefox/whatsnew/whatsnew-115-experiment-eu-vpn.es6.js:26]: 😆

**Review by nathan-barrett — APPROVED** [review]
r+ 🗺️

File: bedrock/firefox/views.py.orig
Comment (craigcook) [reply bedrock/firefox/views.py.orig:1]: doh, that shouldn't have been committed. It was a leftover from a merge conflict.

**Pull request merged**